datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -250,6 +280,29 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn single_distinct_and_grouping_set() -> Result<()> {
```

Comment:
Given there is special handling for CUBE and ROLLUP in this pass, I suggest test coverage for those cases too

Reply:
Yeah, I think there is actually a bug in this. I'll work on a fix.

Reply:
Ok, this optimization is a bit more complicated for grouping sets. We need to create a separate alias for each group. For the moment I have just disabled the optimization for this case.

Reply:
I think disabling the optimization for grouping sets is a wise idea.